presentation-api/controlling-ua/startNewPresentation_success-manual.html

```diff
@@ -21,16 +31,39 @@
     // ---------------------------------------------
     // Start New Presentation Test (success) - begin
     // ---------------------------------------------
-    var startPresentation = function () {
+    presentBtn.onclick = function () {
         presentBtn.disabled = true;
-        promise_test(function () {
+        promise_test(function (t) {
             var request = new PresentationRequest(presentationUrls);
-            return request.start()
-                    .then(function (connection) {
+            return new Promise(function(resolve, reject) {
+                request.onconnectionavailable = t.step_func(function(evt) {
+                    // Check the order #2: "connectionavailable" event fires
+                    assert_equals(order[step], 'FIRE_CONNECTIONAVAILABLE', 'A "connectionavailable" event fires after Promise from PresentationRequest.start() resolves.');
+                    step++;
+                    assert_equals(connection, evt.connection, 'Both Promise from PresentationRequest() and a "connectionavailable" event handler receive the same presentation connection.');
+                });
+
+                request.start()
+                    .then(t.step_func(function (c) {
+                        connection = c;
+
+                        // Check the order #1: Promise from start() resolves
+                        assert_equals(order[step], 'RESOLVE_START', 'Promise from PresentationRequest.start() resolves before connectionavailable event is fired.');
+                        step++;
 
                         // Check the initial state of a presentation connection
                         assert_equals(connection.state, 'connecting', 'The initial state of the presentation connection is "connecting".');
 
+                        // Check the order #3: "connected" event fires
+                        connection.onconnect = function() {
+                            assert_equals(order[step], 'FIRE_CONNECTED', 'A connect event fires after Promise from PresentationRequest.start() resolves and a "connectionavailable" event fires.');
+                            assert_equals(connection.state, 'connected', 'The state of the presentation connection is "connected" when a "connect" event fires.');
+                            resolve();
+                        };
+                        t.step_timeout(function() {
+                            reject('A "connect" event did not fire when the presentation connection was established.')
+                        }, 5000);
+
                         // Check, if the connection ID is set
                         assert_true(!!connection.id, 'The connection ID is set.');
```

Comment:
Could you move these `assert_true` checks together with the check on the initial connection state? Here, they seem to be done later for a specific reason, and I don't think there is any reason to do so in practice.

Also, you can probably drop the comments, these asserts are self-explanatory.